packages/components/src/range-control/utils.js

```diff
@@ -11,35 +11,42 @@ import { useCallback, useRef, useEffect, useState } from '@wordpress/element';
 /**
  * A float supported clamp function for a specific value.
  *
- * @param {number} value The value to clamp
+ * @param {number|null} value The value to clamp
  * @param {number} min The minimum value
  * @param {number} max The maxinum value
  *
  * @return {number} A (float) number
  */
-function floatClamp( value, min, max ) {
+export function floatClamp( value, min, max ) {
+	if ( ! isFinite( value ) ) {
+		return null;
+	}
+
 	return parseFloat( clamp( value, min, max ) );
 }
 
 /**
  * Hook to store a clamped value, derived from props.
  */
-export function useControlledRangeValue( { min, max, value: valueProp = 0 } ) {
+export function useControlledRangeValue( { min, max, value: valueProp } ) {
 	const [ value, setValue ] = useState( floatClamp( valueProp, min, max ) );
 	const valueRef = useRef( value );
 
-	const setClampValue = ( nextValue ) => {
-		setValue( floatClamp( nextValue, min, max ) );
-	};
+	const setClampValue = useCallback(
+		( nextValue ) => {
+			setValue( floatClamp( nextValue, min, max ) );
+		},
+		[ setValue, min, max ]
+	);
 
 	useEffect( () => {
 		if ( valueRef.current !== valueProp ) {
 			setClampValue( valueProp );
 			valueRef.current = valueProp;
 		}
-	}, [ valueProp, setClampValue ] );
+	}, [ valueRef, valueProp, setValue ] );
```

Comment:
I can't immediately recall where the conversation, but I have some recollection of discussing that it may be a concern to include a `useRef` variable as a dependency of another hook?

Reply:
Ah, found it here: https://github.com/WordPress/gutenberg/pull/20219#discussion_r379219934